src/containers/habit-tracker/analytics/year-view/Heatmap.tsx

```diff
@@ -0,0 +1,269 @@
+import ReactApexChart from "react-apexcharts";
+import styled from "styled-components";
+import {
+  colors,
+  heatMapColors,
+} from "../../../../common/utilities/color-utils";
+import { ApexOptions } from "apexcharts";
+
+// Styled component for the Heatmap container card
+const HeatmapCard = styled.div`
+  background-color: ${colors.bgWhite};
+  border-radius: 10px;
+  padding: 10px;
+  width: 900px;
+  max-width: 100%;
+  margin-top: 20px;
+  box-shadow: 0 2px 8px rgba(0, 0, 0, 0.1);
+
+  @media (max-width: 1024px) {
+    width: 600px;
+    padding: 10px;
+  }
+
+  @media (max-width: 768px) {
+    width: 100%;
+    padding: 10px;
+  }
+
+  @media (max-width: 480px) {
+    width: 100%;
+    padding: 10px;
+  }
+`;
+
+// Interface for the heatmap props
+interface HeatmapProps {
+  data: { date: string; count: number }[];
+  year: number;
+  habitName: string;
+}
+
+// Functional component to render the heatmap - used in the year view of the analytics page of the habit tracker
+export function Heatmap({ data, year, habitName }: HeatmapProps) {
+  const daysOfWeek = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"];
+  const weeksInYear = 52;
```

Comment:
These two are global constants and should be at the top of the file with this format:
`DAYS_OF_WEEK` and `WEEKS_IN_YEAR`